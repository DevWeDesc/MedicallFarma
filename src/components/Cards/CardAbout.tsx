"use client";
import React, { useEffect } from "react";
import { ICardAbout } from "../../@types/types";
import { TbTargetArrow } from "react-icons/tb";
import { LiaEyeSolid } from "react-icons/lia";
import { TbCategory2 } from "react-icons/tb";

// "Nossa Miss達o",
// "Nossa Vis達o"
// "Nossas Categorias"

export const CardAbout = ({ title, content }: ICardAbout) => {
  const IconCard = () => {
    const styleIcon = "text-7xl";
    if (title === "Nossa Miss達o") {
      return <TbTargetArrow className={styleIcon} />;
    } else if (title === "Nossa Vis達o") {
      return <LiaEyeSolid className={styleIcon} />;
    } else {
      return <TbCategory2 className={styleIcon} />;
    }
  };

  return (
    <div className="p-16 hover:scale-105 transition-all bg-grayLight flex flex-col gap-4 items-center text-white rounded-lg xl:max-w-sm">
      <IconCard />
      <h2 className="text-2xl font-bold">{title}</h2>
      {typeof content === "object" ? (
        content.map((item, index) => (
          <p key={index} className="opacity-70 font-light text-start w-full">
            {item}
          </p>
        ))
      ) : (
        <p className="opacity-70 font-light">{content}</p>
      )}
    </div>
  );
};
