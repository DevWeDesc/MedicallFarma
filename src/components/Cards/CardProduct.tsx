"use client";
import React, { useEffect, useState } from "react";
import Image from "next/image";
import { IProductsData } from "../../../types/types";

export const CardProduct = ({
  image,
  name,
  content,
  supplier,
  categories,
  size,
}: IProductsData) => {
  // const [size, setConfigCard] = useState("");
  const [contentProduct, setContentProduct] = useState("");

  useEffect(() => {
    size === "large"
      ? setContentProduct(content)
      : setContentProduct(content?.substring(0, 83));
  }, [contentProduct]);

  return (
    <div
      className={`flex hover:scale-105 transition-all flex-col text-left ${
        size === "large"
          ? "gap-4 items-center lg:grid lg:grid-cols-2 lg:justify-items-center"
          : "gap-2"
      } gap-2`}
    >
      <Image
        alt=""
        className={`${size === "large" ? "lg:w-4/5" : ""}`}
        src={image}
      />
      <div className={`${size === "large" && "flex flex-col gap-2"}`}>
        <p
          className={`font-medium min-h-[32px] lg:min-h-[48px] ${
            size === "large" ? "text-2xl lg:text-5xl" : "text-xs lg:text-base"
          }`}
        >
          {name}
        </p>
        <p
          className={`font-light pr-3 ${
            size === "large"
              ? "text-sm opacity-70 lg:text-xl"
              : "text-xs lg:text-sm"
          }`}
        >
          {contentProduct}
        </p>
        <p
          className={`font-medium  ${
            size === "large" ? "text-base lg:text-xl" : "text-xs lg:text-base"
          }`}
        >
          {supplier}
        </p>
        {size === "large" && (
          <div className="mt-4 lg:mt-10 flex flex-col gap-2">
            <button className="w-full rounded-full py-2 lg:text-xl bg-grayLight hover:bg-grayDefault transition-all lg:max-w-[315px] text-white ">
              Solicitar um Or√ßamento
            </button>
            <p className="font-medium lg:text-xl">
              Categorias:{" "}
              {categories?.map((item, index) => {
                let TextCategories =
                  index === categories.length - 1
                    ? item.concat(".")
                    : item.concat(", ");

                return <span key={index}>{TextCategories}</span>;
              })}
            </p>
          </div>
        )}
      </div>
    </div>
  );
};
