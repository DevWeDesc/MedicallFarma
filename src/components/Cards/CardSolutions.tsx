"use client";
import React, { useEffect, useState } from "react";
import Image from "next/image";
import Link from "next/link";
import { ICardSoluctions } from "../../../types/types";

export const CardSolutions = ({ image, title, href }: ICardSoluctions) => {
  const [bgColorCard, setBgColorCard] = useState("");
  useEffect(() => {
    if (title === "Medicamentos") {
      setBgColorCard("bg-blueDark");
    } else if (title === "Dietas Enterais") {
      setBgColorCard("bg-yellowDefault");
    } else {
      setBgColorCard("bg-greenDark");
    }
  }, []);

  return (
    <Link href={href}>
      <div
        className={`relative px-2 pt-3 pb-12 rounded-lg ${bgColorCard} hover:bg-opacity-90 transition-all lg:px-9 lg:py-9`}
      >
        <p className="text-sm font-semibold lg:text-4xl">{title}</p>
        <div className="flex items-start justify-between lg:mt-5">
          <button className="text-[10px] bg-transparent border-b pb-[2px] border-white lg:text-lg">
            Conhe√ßa agora
          </button>
          <Image alt="" className="lg:h-28 lg:w-auto" src={image} />
        </div>
      </div>
    </Link>
  );
};
