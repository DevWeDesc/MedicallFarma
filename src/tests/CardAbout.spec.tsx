/**
 * @jest-environment jsdom
 */
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { CardAbout } from "@/components/Cards/CardAbout";
import "intersection-observer";

describe("BannerAbout render", () => {
  it("renders BannerAbout unchanged", () => {
    const { container } = render(
      <CardAbout content="Descrição do card" title="Titulo do card" />
    );
    expect(container).toMatchSnapshot();
  });

  it("renders BannerAbout text", () => {
    const { getByText } = render(
      <CardAbout content="Descrição do card" title="Titulo do card" />
    );

    expect(getByText("Descrição do card")).toBeInTheDocument();
    expect(getByText("Titulo do card")).toBeInTheDocument();
  });
});
