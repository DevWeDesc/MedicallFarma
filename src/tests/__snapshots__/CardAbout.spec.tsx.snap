// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerAbout render renders BannerAbout unchanged 1`] = `
<div>
  <div
    class="p-16 hover:scale-105 transition-all bg-grayLight flex flex-col gap-4 items-center text-white rounded-lg xl:max-w-sm"
  >
    <svg
      class="text-7xl"
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
        stroke="none"
      />
      <path
        d="M14 4h6v6h-6z"
      />
      <path
        d="M4 14h6v6h-6z"
      />
      <path
        d="M17 17m-3 0a3 3 0 1 0 6 0a3 3 0 1 0 -6 0"
      />
      <path
        d="M7 7m-3 0a3 3 0 1 0 6 0a3 3 0 1 0 -6 0"
      />
    </svg>
    <h2
      class="text-2xl font-bold"
    >
      Titulo do card
    </h2>
    <p
      class="opacity-70 font-light"
    >
      Descrição do card
    </p>
  </div>
</div>
`;
